doc/plugin/DynamicMacros.md

```diff
@@ -0,0 +1,96 @@
+# Kaleidoscope-DynamicMacros
+
+Dynamic macros are similar to [Macros][plugin:macros], but unlike them, they can
+be re-defined without compiling and flashing new firmware: one can change
+dynamic macros via [Focus][plugin:focus], using a tool like
+[Chrysalis][chrysalis].
+
+ [plugin:macros]: Macros.md
+ [plugin:focus]: FocusSerial.md
+ [chrysalis]: https://github.com/keyboardio/Chrysalis
+
+Dynamic macros come with certain limitations, however: unlike the built-in
+macros, dynamic ones do not support running custom code, they can only play back
+a sequence of key events, and do so whenever one presses the dynamic macro key.
+
+You can define up to 32 dynamic macros, there is no limit on their length,
```

Comment:
where is the 32 hardcoded?

Reply:
In [Kaleidoscope-Ranges.h](https://github.com/keyboardio/Kaleidoscope/pull/687/files#diff-767e75f57c447ea243026871575d246bR52-R53). We can easily up the number, mind you.